### PR TITLE
Bug 2109502:  improve bug report links

### PR DESCRIPTION
**Pull request by rhamilto**
Note this fix also partially resolves https://bugzilla.redhat.com/show_bug.cgi?id=2106675, but cannot fully resolve it because the console does not have an existing way to detect if the cluster is a ROSA one.

### CODE REVIEW

File: frontend/public/module/k8s/cluster-settings.ts [hunk 1]
```diff
@@ -240,21 +240,50 @@ export const getReportBugLink = (cv: ClusterVersionKind): { label: string; href:
...
+    case 'azure':
+      productName = 'OpenShift%20Managed%20(Azure)';
+      break;
+    default:
+      productName = 'OKD';
+  }
+  const bugLabel = i18next.t('public~Report bug to Red Hat');
+
+  // Show an OpenShift Container Platform Bugzilla link for OKD
+  // until the OKD product components are updated.
```
Comment (rhamilto) [inline frontend/public/module/k8s/cluster-settings.ts:265]: This is assuming management does in fact want to resume usage of OKD in Bugzilla.

File: frontend/public/module/k8s/cluster-settings.ts [hunk 2]
```diff
@@ -240,21 +240,50 @@ export const getReportBugLink = (cv: ClusterVersionKind): { label: string; href:
...
     ? {
         label: i18next.t('public~Open support case with Red Hat'),
-        href: `https://access.redhat.com/support/cases/#/case/new?product=OpenShift%20Container%20Platform&version=${major}.${minor}&clusterId=${cv.spec.clusterID}`,
+        href: `https://access.redhat.com/support/cases/#/case/new?product=${productName}&version=${major}.${minor}&clusterId=${cv.spec.clusterID}`,
       }
     : {
-        label: i18next.t('public~Report bug to Red Hat'),
-        href: `https://bugzilla.redhat.com/enter_bug.cgi?product=OpenShift%20Container%20Platform&version=${bugzillaVersion}&cf_environment=${environment}`,
+        label: bugLabel,
+        href: `https://issues.redhat.com/secure/CreateIssue.jspa?pid=12332330`,
```
Comment (rhamilto) [inline frontend/public/module/k8s/cluster-settings.ts:286]: @spadgett suggested we may be able to prefill the OpenShift version and perhaps the cluster ID in the Jira issue, but we’d have to investigate if we can set that through a query parameter.  I do not appear to have the ability within Jira to do so however.

Comment (spadgett) [reply frontend/public/module/k8s/cluster-settings.ts:286]: @rhamilto Looks like it is possible: https://confluence.atlassian.com/jirakb/how-to-create-issues-using-direct-html-links-in-jira-server-159474.html

File: frontend/public/module/k8s/cluster-settings.ts [hunk 3]
```diff
@@ -240,21 +240,50 @@ export const getReportBugLink = (cv: ClusterVersionKind): { label: string; href:
...
 
-  // Show a Bugzilla link for prerelease versions and a support case link for supported versions.
   const { major, minor, prerelease } = parsed;
-  const bugzillaVersion = major === 4 && minor <= 3 ? `${major}.${minor}.0` : `${major}.${minor}`;
-  const environment = encodeURIComponent(`Version: ${version}
+  let productName;
+  switch (window.SERVER_FLAGS.branding) {
+    case 'openshift':
+    case 'ocp':
+      productName = 'OpenShift%20Container%20Platform';
```
Comment (spadgett) [inline frontend/public/module/k8s/cluster-settings.ts:248]: I would leave these unencoded

```suggestion
      productName = 'OpenShift Container Platform';
```

then call `encodeURIComponent` when adding them to the URL below.

File: frontend/public/module/k8s/cluster-settings.ts (hunk 2 above)
Comment (rhamilto) [reply frontend/public/module/k8s/cluster-settings.ts]: Turns out it isn't.  :(  See https://github.com/openshift/console/pull/11874/files#diff-ab468773bf6bdc82888f5abe2318066860f9e030113098584bba00ffbace2463R288

File: frontend/public/module/k8s/cluster-settings.ts (hunk 3 above)
Comment (rhamilto) [reply frontend/public/module/k8s/cluster-settings.ts]: Done.

File: frontend/public/module/k8s/cluster-settings.ts [hunk 4]
```diff
@@ -240,21 +240,43 @@ export const getReportBugLink = (cv: ClusterVersionKind): { label: string; href:
...
+      productName = 'OpenShift Container Platform';
+      break;
+    case 'online':
+      productName = 'OpenShift Online';
+      break;
+    case 'dedicated':
+      productName = 'OpenShift Dedicated';
+      break;
+    case 'azure':
+      productName = 'Azure Red Hat OpenShift';
```
Comment (rhamilto) [inline frontend/public/module/k8s/cluster-settings.ts:257]: Depends on https://issues.redhat.com/browse/PCM-11545 being completed

File: frontend/public/module/k8s/cluster-settings.ts [hunk 5]
```diff
@@ -240,21 +240,43 @@ export const getReportBugLink = (cv: ClusterVersionKind): { label: string; href:
...
+      break;
+    default:
+      productName = 'OKD';
+  }
+
+  // Do not show a link for OKD until the new OKD Jira project is ready.
```
Comment (rhamilto) [inline frontend/public/module/k8s/cluster-settings.ts:263]: Waiting for confirmation this is the desired behavior.

Comment (rhamilto) [reply frontend/public/module/k8s/cluster-settings.ts:263]: Confirmed by Michelle Krejci.  See https://coreos.slack.com/archives/C6A3NV5J9/p1658845554775669?thread_ts=1658754855.051769&cid=C6A3NV5J9

**Comment by rhamilto** [comment]
@spadgett, since the Jira go-live is tomorrow, would you prefer I modify this PR to not change the Azure product name until https://issues.redhat.com/browse/PCM-11545 is complete?

**Comment by spadgett** [comment]
> @spadgett, since the Jira go-live is tomorrow, would you prefer I modify this PR to not change the Azure product name until https://issues.redhat.com/browse/PCM-11545 is complete?

@rhamilto Yeah, I would address that in a follow up.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
/hold

need to change `Azure Red Hat OpenShift` to `OpenShift Managed (Azure)` until https://issues.redhat.com/browse/CPCCM-9926 is complete

**Comment by rhamilto** [comment]
/hold cancel

@spadgett, please retag.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**